linkerd/app/gateway/src/lib.rs

```diff
@@ -84,7 +84,7 @@ where
     O::Response:
         io::AsyncRead + io::AsyncWrite + tls::HasNegotiatedProtocol + Send + Unpin + 'static,
     O::Future: Send + Unpin + 'static,
-    P: profiles::GetProfile<profiles::LogicalAddr> + Clone + Send + Sync + Unpin + 'static,
+    P: profiles::GetProfile<profiles::LookupAddr> + Clone + Send + Sync + Unpin + 'static,
```

Comment:
AFAICT, there are now no impls of `GetProfile` for anything other than a `LookupAddr`:
```bash
:# eliza on butterfly in linkerd2-proxy on  ver/profile-addr [$?] via ⚙️ v1.49.0 
:; rg "GetProfile<"
linkerd/service-profiles/src/default.rs
25:    S: GetProfile<T>,

linkerd/service-profiles/src/discover.rs
14:    G: GetProfile<F::Request> + Clone,
34:    G: GetProfile<T>,

linkerd/service-profiles/src/lib.rs
54:pub trait GetProfile<T> {
68:impl<T, S> GetProfile<T> for S
84:    P: GetProfile<T>,

linkerd/app/outbound/src/discover.rs
28:        P: profiles::GetProfile<profiles::LookupAddr> + Clone + Send + 'static,

linkerd/app/outbound/src/lib.rs
113:        P: profiles::GetProfile<profiles::LookupAddr> + Clone + Send + 'static,
141:        P: profiles::GetProfile<profiles::LookupAddr> + Clone + Send + Sync + Unpin + 'static,

linkerd/app/outbound/src/ingress.rs
44:        P: profiles::GetProfile<profiles::LookupAddr> + Clone + Send + Sync + Unpin + 'static,

linkerd/app/gateway/src/lib.rs
87:    P: profiles::GetProfile<profiles::LookupAddr> + Clone + Send + Sync + Unpin + 'static,

linkerd/app/inbound/src/lib.rs
138:        P: profiles::GetProfile<profiles::LookupAddr> + Clone + Send + Sync + 'static,
228:        P: profiles::GetProfile<profiles::LookupAddr> + Clone + Send + Sync + 'static,

linkerd/app/inbound/src/http/mod.rs
118:        P: profiles::GetProfile<profiles::LookupAddr> + Clone + Send + Sync + 'static,

```

Should we maybe just remove the type parameter and change `GetProfile` to
```rust
pub trait GetProfile {
    type Error: Into<Error>;
    type Future: Future<Output = Result<Option<Receiver>, Self::Error>>;

    fn get_profile(&mut self, target: LookupAddr) -> Self::Future;
}
```
or am I missing something?

Reply:
seems like a fine change but it will touch more than this pr needs to.

Reply:
👍 i can throw together a PR after this lands